### DISCUSSION TITLE
having problems with ww3 setup with ww3 7.14

### DISCUSSION THREAD

**Discussion by JMSnyder1980** [NOAA-EMC/WW3#893] [General]
I am trying to run w3_automake do compile ww3 7.14 but all I get is this message:

./w3_automake 
 
                *****************************
              ***   WAVEWATCH III automake  ***
                *****************************
 
 Exporting WWATCH3_ENV=/home/ww3/WW3/model/bin/wwatch3.env
Shared Object     : No
 
[INFO] list of the programs selected : ww3_strt ww3_bound ww3_outf ww3_outp ww3_trck ww3_grib ww3_gint gx_outf gx_outp ww3_uprstr ww3_systrk ww3_gspl ww3_prep ww3_shel
 
[INFO] SEQ progs :  ww3_strt ww3_bound ww3_outf ww3_outp ww3_trck ww3_grib ww3_gint gx_outf gx_outp ww3_uprstr ww3_systrk ww3_gspl ww3_prep  ww3_shel
 
 
                *****************************
              ***   WAVEWATCH III make      ***
                *****************************
 
Main directory    : /home/ww3/WW3/model
Scratch directory : /home/ww3/WW3/model/tmp
Save source codes : yes
Save listings     : yes
Shared Object     : No
 
[INFO] list of the programs selected : ww3_strt ww3_bound ww3_outf ww3_outp ww3_trck ww3_grib ww3_gint gx_outf gx_outp ww3_uprstr ww3_systrk ww3_gspl ww3_prep ww3_shel
Making makefile ...
 
                *****************************
              ***   WAVEWATCH III makefile  ***
                *****************************
 
switch = /home/ww3/WW3/model/work_SEQ/switch.SEQ
   Checking all subroutines for modules (this may take a while) ...

The w3_automake and w3_make stop at the point where it says "Checking all subroutines for modules (this may take a while)..." without any indication as to why the setup crashed.   I following what it says in the quick start guide and ww3 manual exactly.  Does anyone know what is going on and how to resolve this issue?  

When I checked the exe directory all I get are these files:

ls ../exe/
switch  switch.SEQ

**Comment by aliabdolali** [comment]
Automake is not fully functional, and gnu build system is about to retire. In stead, try [cmake](https://github.com/NOAA-EMC/WW3/tree/develop/model#readme), it does the same as automake but more efficient. It is tested extensively.

**Reply by JMSnyder1980** [reply]
how do I use cmake?

**Reply by aliabdolali** [reply]
follow https://github.com/NOAA-EMC/WW3/tree/develop/model#readme

**Reply by JMSnyder1980** [reply]
I did do the cmake and make install but get this netcdf error:

/home/ww3/WW3/model/src/SCRIP/scrip_netcdfmod.f90:19:6:

   use netcdf
      1
Fatal Error: Can't open module file ‘netcdf.mod’ for reading at (1): No such file or directory
compilation terminated.
make[2]: *** [model/src/CMakeFiles/ww3_lib.dir/build.make:998: model/src/CMakeFiles/ww3_lib.dir/SCRIP/scrip_netcdfmod.f90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:285: model/src/CMakeFiles/ww3_lib.dir/all] Error 2
make: *** [Makefile:136: all] Error 2


How do I resolve this?

**Reply by aliabdolali** [reply]
you need to have netcdf libraries installed on your system. You can exclude the programs which are netcdf dependent:
_NetCDF - Optionally searched for to build NetCDF executables. Required when SCRIPNC, TRKNC, or OASIS are in switch list._

There is section in the instruction about it, please check the entire instruction:

_Finding Libraries_
_CMake has a standardized way of searching for external libraries including default system paths such as /usr/local._

_If a library is located in a non-standard location CMake will search multiple locations including CMAKE_PREFIX_PATH (a semi-colon separated list) or <name>_ROOT._

_Note: If your system provides the paths to libraries and include directories automatically (e.g. via a compiler wrapper script, as is the case with the Cray Compiler Environment), then you might wish to stop CMake from trying to automatically finding particular libraries. This can be achieved by passing a list of libraries to ignore via the EXCLUDE_FIND option to CMake (currently only implemented for the netCDF library). E.g:_

`cmake .. -DEXCLUDE_FIND="netcdf"`

**Comment by JMSnyder1980** [comment]
I took out the netcdf dependent SCRIPNC from the switch_NCEP_gwm file in WW3/model/bin directory

I ran:

cmake .. -DNETCDF=OFF -DEXCLUDE_FIND="netcdf" -DSWITCH=/home/ww3/WW3/model/bin/switch_NCEP_gwm -DCMAKE_INSTALL_PREFIX=install

and make install

I found in WW3/build/bin executables and when I ran ww3_shel from there I got the following error:


[ww3@wavewatch bin]$ ./ww3_shel

                     *** WAVEWATCH III Program shell ***      
               ===============================================

At line 1195 of file /home/ww3/WW3/model/src/ww3_shel.F90 (unit = 11, file = 'fort.11')
Fortran runtime error: End of file

Error termination. Backtrace:
#0  0x7fdd41449171 in ???
#1  0x7fdd41449d19 in ???
#2  0x7fdd4144a521 in ???
#3  0x7fdd416521bb in ???
#4  0x7fdd416527b2 in ???
#5  0x7fdd4164f33b in ???
#6  0x7fdd41653ec4 in ???
#7  0x7fdd41654d34 in ???
#8  0x402570 in w3shel
	at /home/ww3/WW3/model/src/ww3_shel.F90:1195
#9  0x401e4c in main
	at /home/ww3/WW3/model/src/ww3_shel.F90:253


This type of output indicates that the compilation did not go through properly for ww3_shel. 


I did see this warning in the compilation:

[ 55%] Building Fortran object model/src/CMakeFiles/ww3_lib.dir/w3srcemd.F90.o
/home/ww3/WW3/model/src/w3srcemd.F90:1675:0:

 #endif W3_PDLIB
 
Warning: extra tokens at end of #endif directive


 Can you help me resolve this issue?
